### ISSUE TITLE
RPM packages for mofed 5.x depend on specific versions of mofed

### ISSUE THREAD

**Issue by abellina** [openucx/ucx#6662]
I am trying to install the UCX 1.10 RPM packages in centos 7.9 with MLNX_OFED:
```
$ ofed_info -n
5.3-1.0.0.1
```

But I see:
```
--> Finished Dependency Resolution
Error: Package: ucx-1.10.0-1.el7.x86_64 (/ucx-1.10.0-1.el7.x86_64)
           Requires: libibverbs = 50mlnx1-1.50100.0
           Installed: libibverbs-52mlnx1-1.53100.x86_64 (installed)
               libibverbs = 52mlnx1-1.53100
           Available: libibverbs-22.4-5.el7.i686 (base)
               libibverbs = 22.4-5.el7
 You could try using --skip-broken to work around the problem
```

Not entirely sure why the `ucx` package requires libibverbs in the first place (I'd assume that to be the `ucx-ib` and `ucx-rdmacm` packages), so that's part of it. We want to be able to install `ucx`, even if MLNX_OFED is not installed.

`ucx-cuda` requires the CUDA runtime installed. This makes sense, but not all installations are via `rpm`, so that may be another issue. Perhaps we should relax this requirement, like https://github.com/openucx/ucx/pull/5776.

```
Error: Package: ucx-cuda-1.10.0-1.el7.x86_64 (/ucx-cuda-1.10.0-1.el7.x86_64)
           Requires: ucx(x86-64) = 1.10.0-1.el7
           Available: ucx-1.5.2-1.el7.x86_64 (base)
               ucx(x86-64) = 1.5.2-1.el7
Error: Package: ucx-cuda-1.10.0-1.el7.x86_64 (/ucx-cuda-1.10.0-1.el7.x86_64)
           Requires: libcudart.so.11.0()(64bit)
Error: Package: ucx-cuda-1.10.0-1.el7.x86_64 (/ucx-cuda-1.10.0-1.el7.x86_64)
           Requires: libcuda.so.1()(64bit)
Error: Package: ucx-cuda-1.10.0-1.el7.x86_64 (/ucx-cuda-1.10.0-1.el7.x86_64)
           Requires: libcudart.so.11.0(libcudart.so.11.0)(64bit)
Error: Package: ucx-cuda-1.10.0-1.el7.x86_64 (/ucx-cuda-1.10.0-1.el7.x86_64)
           Requires: ucx(x86-64) = 1.10.0-1.el7
           Installing: ucx-1.5.2-1.el7.x86_64 (base)
               ucx(x86-64) = 1.5.2-1.el7
```

**Comment by yosefe** [comment]
> I am trying to install the UCX 1.10 RPM packages in centos 7.9 with MLNX_OFED:
> 
> ```
> $ ofed_info -n
> 5.3-1.0.0.1
> ```

Seems 2 issues here:
1. UCX requires a strict version of MLNX_OFED/libibverbs, but it should require '>='
2. Verbs dependencies "crawled" into main UCX package, whereas they should have been only in ucx-ib subpackage

> ucx-cuda requires the CUDA runtime installed. This makes sense, but not all installations are via rpm, so that may be another issue. Perhaps we should relax this requirement, like #5776.

When we install UCX from RPM, all the dependencies should also come from RPM. If this is not the case on a particular system, then should build and install UCX from source tarball. This is a standard way with Linux packages.

**Comment by yosefe** [comment]
Update on issue (1): strict verbs dependency was added as part of https://github.com/openucx/ucx/pull/5184
now that wit use backward compatible MOFED versions, it's not needed anymore

**Comment by abellina** [comment]
> When we install UCX from RPM, all the dependencies should also come from RPM. If this is not the case on a particular system, then should build and install UCX from source tarball. This is a standard way with Linux packages.

Agree on this 100%. After installing CUDA via RPM, the cuda part of it goes away. So back to the main issue (on the IB requirement)..

With the package the way it is (requiring 5.0-1) not only do I need to install MLNX_OFED for a system that may not need it, but it also forces me to pick a specific version of the driver, and in this case (Centos 7.9) there isn't a MLNX_OFED 5.0-1. This is likely due to the release date of Centos 7.9 (looks like November 2020, which is after MLNX_OFED 5.0-1), further complicating things.

I also looked at the rhel repo under (https://linux.mellanox.com/public/repo/mlnx_ofed/latest/rhel7.9/x86_64/). This is almost what we want (there are rhel 7.9 rpms here), but there is only one CUDA runtime for 10.2, so I couldn't install it with 11.2 as the only run time in the system. I also tried picking up the `ucx-cuda` for CUDA 11.2 from github and trying to add it after installing `ucx` from the mellanox rhel repo, but `ucx-cuda` from github requires a specific `ucx`, and so I can't play that trick.

**Comment by yosefe** [comment]
> With the package the way it is (requiring 5.0-1) not only do I need to install MLNX_OFED for a system that may not need it, but it also forces me to pick a specific version of the driver, and in this case (Centos 7.9) there isn't a MLNX_OFED 5.0-1. This is likely due to the release date of Centos 7.9 (looks like November 2020, which is after MLNX_OFED 5.0-1), further complicating things.

This is leftover from the days of MLNX_OFED < 5.0 and is not what we want. #6667 should fix it
